File: SampleApps/ObjC/AdColonyInstantFeed/AdColonyInstantFeed/FeedCellAd.m
//
//  FeedCellAd.m
//  AdColonyInstantFeed
//
//  Created by Owain Moss on 2/4/15.
//  Copyright (c) 2015 AdColony. All rights reserved.
//

#import "FeedCellAd.h"

@implementation FeedCellAd

#pragma mark - Video View

- (void)setAdView:(AdColonyNativeAdView *)adView {
    [_adContainer addSubview:adView];
    
    _adView = adView;
}

#pragma mark - Pause/Resume

- (void)pause {
    [_adView pause];
}

- (void)resume {
    [_adView resume];
}
@end

File: SampleApps/Swift/AdColonyInstantFeed/AdColonyInstantFeed/FeedCellAd.swift
//
//  FeedCellAd.swift
//  AdColonyInstantFeed
//
//  Created by Owain Moss on 2/11/15.
//  Copyright (c) 2015 AdColony. All rights reserved.
//

import Foundation
import UIKit

class FeedCellAd: UITableViewCell
{
    @IBOutlet weak var iconView: UIImageView!
    @IBOutlet weak var advertiserLabel: UILabel!
    @IBOutlet weak var titleLabel: UILabel!
    @IBOutlet weak var adContainer: UIView!
    
    var adView: AdColonyNativeAdView? {
        willSet(newAdView) {
            if let view = newAdView as AdColonyNativeAdView? {
               adContainer.addSubview(view)
            }
        }
    }
    
    //===================================================
    // MARK:- Pause/Resume
    //===================================================
    
    func pause() -> Void
    {
        adView?.pause()
    }
    
    func resume() -> Void
    {
        adView?.resume()
    }
}
